src/qibo/tomography/gate_set_tomography.py

```diff
@@ -0,0 +1,371 @@
+        nshots (int, optional): Number of shots used in Gate Set Tomography.
+        noise_model (:class:`qibo.noise.NoiseModel`, optional): Noise model applied to simulate noisy computation.
+        include_empty (bool, optional): If ``False``, only perform gate set tomography for the list of gates in ``gate_set``.
+            If ``True``, perform gate set tomography for the list of gates in ``gate_set`` and also empty circuits.
+        invert_register (bool, optional): If ``True``, one needs to specify which qubit(s) to reset.
+        Pauli_Liouville (bool, optional): If ``True``, returns gate set tomography of the gates in the Pauli-Liouville representation.
+        backend (:class:`qibo.backends.abstract.Backend`, optional): Calculation engine.
+    Returns:
+        list(ndarray): List of matrices of the gate(s) in gate set tomography.
+    """
+    matrices = []
+    empty_matrices = []
+    for nqubits in range(1, 3):
+        empty_matrix = _gate_set_tomography(
+            nqubits=nqubits,
+            gate=None,
+            nshots=nshots,
+            noise_model=noise_model,
+            invert_register=invert_register,
+            backend=backend,
+        )
+        empty_matrices.append(empty_matrix)
+        if include_empty:
+            matrices.append(empty_matrix)
+
+    for gate in gate_set:
+        if gate is not None:
+            init_args = signature(gate).parameters
+            if "q" in init_args:
+                nqubits = 1
+            elif "q0" in init_args and "q1" in init_args and "q2" not in init_args:
+                nqubits = 2
+            else:
+                raise_error(
+                    RuntimeError,
+                    f"Gate {gate} is not supported for `GST`, only 1- and 2-qubits gates are supported.",
+                )
+            gate = gate(*range(nqubits))
+
+        if Pauli_Liouville is False:
+            matrices.append(
+                _gate_set_tomography(
+                    nqubits=nqubits,
+                    gate=gate,
+                    nshots=nshots,
+                    noise_model=noise_model,
+                    invert_register=invert_register,
+                    backend=backend,
+                )
+            )
+        elif Pauli_Liouville is True:
+            T = np.array([[1, 1, 1, 1], [0, 0, 1, 0], [0, 0, 0, 1], [1, -1, 0, 0]])
+            matrix = _gate_set_tomography(
+                nqubits=nqubits,
+                gate=gate,
+                nshots=nshots,
+                noise_model=noise_model,
+                invert_register=invert_register,
+                backend=backend,
+            )
+            T_matrix = T if matrix.shape[0] == 4 else np.kron(T, T)
+            empty = empty_matrices[0] if matrix.shape[0] == 4 else empty_matrices[1]
+
+            Pauli_Liouville_form = (
+                T_matrix @ np.linalg.inv(empty) @ matrix @ np.linalg.inv(T_matrix)
+            )
+            matrices.append(Pauli_Liouville_form)
```

Comment:
```suggestion
        matrices.append(
            _gate_set_tomography(
                nqubits=nqubits,
                gate=gate,
                nshots=nshots,
                noise_model=noise_model,
                invert_register=invert_register,
                backend=backend,
            )
        )
        if Pauli_Liouville:
            T = backend.np.array([[1, 1, 1, 1], [0, 0, 1, 0], [0, 0, 0, 1], [1, -1, 0, 0]])
            T_matrix = T if matrix.shape[0] == 4 else backend.np.kron(T, T)
            empty = empty_matrices[0] if matrix.shape[0] == 4 else empty_matrices[1]

            matrices = [
                T_matrix @ np.linalg.inv(empty) @ matrix @ np.linalg.inv(T_matrix)
                for matrix in matrices
            ]
```
you don't need to repeat the code calling `_gate_set_tomography`